ravenml/data/write_dataset.py

```diff
@@ -0,0 +1,248 @@
+import os, shutil, time, json
+import contextlib2
+from pathlib import Path
+from object_detection.dataset_tools import tf_record_creation_util
+from random import shuffle
+from datetime import datetime
+from ravenml.data.interfaces import CreateInput
+from ravenml.utils.question import cli_spinner, cli_spinner_wrapper
+from ravenml.utils.config import get_config
+from ravenml.data.helpers import filter_sets, load_metadata, ingest_metadata
+from ravenml.utils.io_utils import download_data_from_s3, copy_data_locally
+
+
+class DatasetWriter(object):
+
+    def __init__(self, create: CreateInput, **kwargs):
+        self.associated_files = kwargs['associated_files']
+        self.related_data_prefixes = kwargs['related_data_prefixes']
+        
+        metadata = create.metadata
+        self.num_folds = metadata['kfolds'] if metadata.get('kfolds') else 5 # Never actually used
+        self.test_percent = metadata['test_percent'] if metadata.get('test_percent') else .2
+        self.label_to_int_dict = {}
+        self.image_ids = None
+        self.filter_metadata = None
+        self.temp_dir = create.plugin_metadata['temp_dir_path']
+        self.dataset_path = create.dataset_path
+        self.dataset_name = metadata['dataset_name']
+        self.created_by = metadata['created_by']
+        self.comments = metadata['comments']
+        self.plugin_name = create.plugin_metadata['architecture']
+        self.imagesets_used = metadata['imagesets_used']
+        self.filter = metadata['filter']
+        self.cache = create.plugin_cache
+    
+    def construct_all(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def export_as_TFExample(self, *args, **kwargs):
```

Comment:
is it possible to set up this interface to work with ANY library, not just TensorFlow? I realize we only use TF for the moment, but ideally ravenML is library-independent. having a required function named "export_as_TFExample" ties us to TF.